src/app/helpers/formTemplate/formTemplate.ts

```diff
@@ -0,0 +1,118 @@
+import { OnInit } from "@angular/core";
+import { ActivatedRoute, Router } from "@angular/router";
+import { ToastrService } from "ngx-toastr";
+import { Observable } from "rxjs";
+import { takeUntil } from "rxjs/operators";
+import { WithFormCheck } from "src/app/guards/form/form.guard";
+import { AbstractModel } from "src/app/models/AbstractModel";
+import { ApiErrorDetails } from "src/app/services/baw-api/api.interceptor.service";
+import { ResolvedModel } from "src/app/services/baw-api/resolver-common";
+import { PageComponent } from "../page/pageComponent";
+
+export abstract class FormTemplate<M extends AbstractModel, F>
+  extends WithFormCheck(PageComponent)
+  implements OnInit {
+  public loading: boolean;
+  public models: { [key: string]: AbstractModel };
+  public schema: FormlySchema;
+
+  /**
+   * @param modelKeys Model resolver keys
+   * @param api API Service
+   * @param notifications ToastrService
+   * @param route ActivatedRoute
+   * @param router Router
+   */
+  constructor(
+    protected modelKeys: string[],
+    protected fields: any[],
+    protected notifications: ToastrService,
+    protected route: ActivatedRoute,
+    protected router: Router
+  ) {
+    super();
+  }
+
+  ngOnInit() {
+    // Grab models from route data
+    const data = this.route.snapshot.data;
+    const resolvedModels: ResolvedModel<AbstractModel>[] = this.modelKeys.map(
+      key => data[key]
+    );
+
+    // If any models failed, return
+    if (resolvedModels.some(resolvedModel => resolvedModel.error)) {
+      return;
```

Comment:
what is the effect of returning early here? Is this component broken? Disabled?

Reply:
Yes, this returns before any of the variables are initialised so that the parent component knows not to render anything. I can make it set a `error` variable or something instead to make this behaviour more obvious.